Explorer/Assets/DCL/Profiles/Avatar.cs

```diff
@@ -0,0 +1,19 @@
+using CommunicationData.URLHelpers;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL.Profiles
+{
+    public class Avatar
```

Comment:
I believe this class should not be fully open: 
- Collections should be publicly exposed as `IReadOnly`
- There should be no public setters: consider introducing a constructor and/or making them `internal`

The same applies to other classes: `Profile`, `Emote`

Reply:
We should be prepared for memory clean-up/re-utilization.

- Please make these classes poolable: preallocate their collections accordingly
- Expect that at some point they won't be kept in the cache forever: thus, they will be returned to the pool

It is easier to make these changes in the beginning than to adjust every single class in the future.

Reply:
Collections set as read only and removed public setters. I did some pool management on dtos, but i find that profiles might potentially grow over time anyway. What/when is the reason for disposal?

Reply:
>What/when is the reason for disposal?

Cache can't grow indefinitely, we will apply some limits and tactics as @popuz did with Memory Budgeting.

We should use something similar to detect if `Profile` is not referenced and dispose of them when there is memory pressure.

Reply:
For now, the system is simple, so it should be as easier as presenting Cache with `Unload` method, then `Register` it to the `CacheCleaner` and call it in the `CacheCleaner.Unload` method. The only thing you will need to care is proper Referencing/Dereferncing, but I don't see any dependencies to already loaded Wearables, Gltf's, AssetBudles or Textures, so it seems to be also relatively easy to achieve. 
Yesterday I finished [documentation](https://github.com/decentraland/unity-explorer/wiki/Memory-Budgeting-&-Resource-Unloading), so you can take a look for more info if needed

Reply:
I hooked the cache into the `CacheCleaner` however i dont see yet a clear strategy for referencing/dereferencing profiles, since they are going to be used in different features (like avatar shapes, socials, backpack, passport, etc)